calculate_costs.py

```diff
@@ -0,0 +1,870 @@
+            find_prices(strategy, voltage_level, utilization_time_per_year_flex,
+                        energy_supply_per_year, UTILIZATION_TIME_PER_YEAR_EC)
+
+        # commodity costs for flexible load:
+        price_list_flex_load = [commodity_charge_flex] * len(power_flex_load_list)
+        commodity_costs_eur_per_year_flex, commodity_costs_eur_sim_flex = \
+            calculate_commodity_costs(price_list_flex_load, power_flex_load_list,
+                                      timestep_s, duration_year_s)
+
+        # capacity costs for flexible load:
+        power_outside_core_standing_time_flex_list = []
+        # find times of grid supply outside of core standing time
+        for i in range(len(timestamps_list)):
+            # not within core standing time (cst)
+            if (not dt_within_core_standing_time(timestamps_list[i], core_standing_time_dict) and
+                    power_flex_load_list[i] < 0):
+                power_outside_core_standing_time_flex_list.append(power_flex_load_list[i])
+        # charging only within core standing time (cst)
+        if power_outside_core_standing_time_flex_list == []:
+            max_power_grid_supply_outside_cst_flex = 0
+        else:
+            max_power_grid_supply_outside_cst_flex = min(power_outside_core_standing_time_flex_list)
+
+        capacity_costs_eur_per_year_flex, capacity_costs_eur_sim_flex = \
+            calculate_capacity_costs_jlp(capacity_charge_flex,
+                                         max_power_grid_supply_outside_cst_flex,
+                                         timestamps_list, timestep_s, duration_year_s)
+
+        # TOTAl COSTS:
+        commodity_costs_eur_sim = commodity_costs_eur_sim_fix + commodity_costs_eur_sim_flex
+        commodity_costs_eur_per_year = (commodity_costs_eur_per_year_fix
+                                        + commodity_costs_eur_per_year_flex)
+        capacity_costs_eur_sim = capacity_costs_eur_sim_fix + capacity_costs_eur_sim_flex
+        capacity_costs_eur_per_year = (capacity_costs_eur_per_year_fix
+                                       + capacity_costs_eur_per_year_flex)
+
+    # COSTS NOT RELATED TO STRATEGIES
+
+    # ADDITIONAL COSTS FOR JLP-CONSUMERS:
+    if fee_type == "jlp":
+        additional_costs_per_year = price_sheet["grid_fee"]["jlp"]["additional_costs"]["costs"]
+        additional_costs_sim = additional_costs_per_year * (duration_sim_s / duration_year_s)
+    else:
+        additional_costs_per_year = 0
+        additional_costs_sim = 0
+
+    # COSTS FOR POWER PROCUREMENT:
+    power_procurement_charge = price_sheet["power_procurement"]["charge"]  # [ct/kWh]
+    power_procurement_costs_sim = (power_procurement_charge * energy_supply_sim / 100)  # [EUR]
+    power_procurement_costs_per_year = (power_procurement_costs_sim
+                                        * (duration_year_s / duration_sim_s))  # [EUR]
+
+    # COSTS FROM LEVIES:
+
```

Comment:
This whole cost calculation can be done easier by iterating through the names (keys) of levies in the price sheet. Then there might also be no problem if people change names of levies in the price_sheet.json.